### PR TITLE
fix(bigquery): Update #ddl? to support ALTER TABLE

### PR DESCRIPTION
**Pull request by quartzmo**
* Update Bigquery::Data#ddl?
* Update Bigquery::QueryJob#ddl?

closes: #16097

### CODE REVIEW

File: google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb [hunk 1]
```diff
@@ -349,4 +349,125 @@
...
+    it "knows its DDL CREATE_VIEW statement type" do
+      gapi = query_job_resp_gapi "query is ignored", statement_type: "CREATE_VIEW"
+      data = Google::Cloud::Bigquery::Data.from_gapi_json data_hash, nil, gapi, nil
+
+      _(data.statement_type).must_equal "CREATE_VIEW"
+      _(data.ddl?).must_equal true
+      _(data.dml?).must_equal false
+    end
+
+    it "knows its DDL \n statement type" do
```
Comment (dazuma) [inline google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb:401]: Seems odd for a statement type to be the newline character. Are we sure this is correct? (Even if so, I suggest escaping the backslash in this string so there isn't a literal newline in the test name.)

Comment (quartzmo) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb:401]: It does seem strange but it is the existing behavior--although that behavior was lacking an explicit test!

I'll try to track down the reason for it.

Comment (quartzmo) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb]: Escaped the newline character in the test names.

Comment (shollyman) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb]: Went and poked at this.  In the broken ALTER TABLE case, `ddlOperationPerformed` is simply missing while `statementType` is set to `ALTER_TABLE`.  I'm not finding an instance of newline values in ddlOperationPerformed or in statementType.

Comment (quartzmo) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb]: @shollyman So can this PR can safely remove the newline character from the `ddl?` [statement types list](https://github.com/googleapis/google-cloud-ruby/blob/8f02c18a3b32025758f1bdfffd23fd9e2077d732/google-cloud-bigquery/lib/google/cloud/bigquery/data.rb#L287)?

**Review by shollyman — APPROVED** [review]

File: google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb (hunk 1 above)
Comment (shollyman) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb]: Yep, approved

Comment (quartzmo) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb]: OK, I'll update the PR to remove the newline character from the list. Thank you!

Comment (quartzmo) [reply google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb]: @dazuma I removed the newline character (`\n`) from the `#ddl?` methods' statement types, and their tests.

**Review by dazuma — APPROVED** [review]

**Pull request merged**